### PR TITLE
Use D-Bus activation instead of explicit service starting

### PR DESCRIPTION
**Pull request by mvidner**
## Problem

- https://github.com/yast/d-installer/issues/287

> The D-Installer backend is composed of several D-Bus services. A service might have dependencies on other services, so they are started sequentially (as we know the dependencies in advance). We decided to use a [simplistic approach](https://github.com/yast/d-installer/blob/master/service/bin/d-installer#L52-L58) (waiting 1 second between one service and the next), but it is time to evolve this solution to something more robust. If some service takes more time than expected to start, it might cause the whole start-up sequence to fail.

The startup failure then shows as "**Cannot connect to D-Bus**" in the web UI, as reported in #280 and #286

## Solution

D-Bus has an [activation mechanism][dba] exactly for this, now we use it.

- As we now ship a service running as root that users could potentially call and misuse, an audit is required. ~~Filed [bsc#1205246](https://bugzilla.suse.com/show_bug.cgi?id=1205246)~~ Duplicate of [bsc#1202059](https://bugzilla.suse.com/show_bug.cgi?id=1202059) AUDIT-0: ruby3.1-rubygem-d-installer: review of D-Bus services org.opensuse.DInstaller (and friends)

Also I've kept the manual running mechanism, by increasing the inter-service delay to 5 seconds and adding a tiny process manager with a help:
```console
$ sudo bundle exec bin/d-installer -h
bin/d-installer -h|--help
bin/d-installer -s|--status
bin/d-installer -k|--kill
bin/d-installer language|manager|questions|software|storage|users
```

[dba]: https://dbus.freedesktop.org/doc/dbus-specification.html#message-bus-starting-services

- [x] development: how to terminate it, Ctrl-C no longer easy. See also https://github.com/yast/d-installer/issues/206
- [x] adjust README talking about manually starting it
- [x] clarify relationship between systemd and D-Bus `*.service`
- [x] syslog: adjust logger not to buffer and to omit the timestamp
- [x] integrate the setup into `setup.sh`

## Testing

- The changes are at the application+integration level so there are no unit tests
- Manually tested these deployment use cases:
    - development from git checkout: by running `setup.sh`
    - production, on [YaST:Head:D-Installer/d-installer-live/images](https://build.opensuse.org/package/binaries/YaST:Head:D-Installer/d-installer-live/images): did not actually test those, only built a *.gem and then a rubygem-d-installer.rpm and checked that the *.service files are present

## Screenshots

*If the fix affects the UI attach some screenshots here.*

### CODE REVIEW

**Comment by imobachgs** [comment]
It looks great! A huge improvement over what we had. Thanks!

**Comment by mvidner** [comment]
OK, the wildcard in `gem2rpm.yml` works as I intended and a RPM builds correctly, with a pleasantly helpful report about the D-Bus services needing a security review, and some policy linting:

```
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller.Language"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller.Questions"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller.Software"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller.Storage"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
ruby3.1-rubygem-d-installer.x86_64: W: dbus-policy-allow-receive <allow receive_sender="org.opensuse.DInstaller.Users"/> /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf
allow receive_* is normally not needed as that is the default.

ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.Language.service (sha256 ...)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.Software.service (sha256 ...)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.Questions.service (sha256 ...)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.service (sha256 ...)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.Users.service (sha256 ...)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system-services/org.opensuse.DInstaller.Storage.service (sha256 file digest default filter:fa26ad5580054daaea6d6c34f6367ee862f0ab5aa690b00b8d7b32c522d85124 shell filter:fa26ad5580054daaea6d6c34f6367ee862f0ab5aa690b00b8d7b32c522d85124 xml filter:<failed-to-calculate>)
ruby3.1-rubygem-d-installer.x86_64: E: dbus-file-unauthorized (Badness: 10) /usr/share/dbus-1/system.d/org.opensuse.DInstaller.conf (sha256 file digest default filter:a91a92809447e2212a37aa9905ef5023020decec9cc02403cb1a6673aba85364 shell filter:34e8141d51ceb164bf802222da1d91304a42f61e43b1d24defdd152da8f1eaa7 xml filter:73b62795796f9fe48e3e8dea251196014615b01e571958d65c427cf95aeaa907)
Packaging D-Bus services requires a review and whitelisting by the SUSE
security team. If the package is intended for inclusion in any SUSE product
please open a bug report to request review of the package by the security
team. Please refer to
https://en.opensuse.org/openSUSE:Package_security_guidelines#audit_bugs for
more information.
```

File: README.md [hunk 1]
```diff
@@ -101,8 +101,11 @@ $ make devel-install
 $ cd -
 ~~~
 
-* Start the services: beware that D-Installer must run as root (like YaST does) to do hardware probing, partition the disks, install the software and so on.
-
+* Start the services:
```
Comment (mvidner) [inline README.md:104]: BTW the instructions are getting complicated, maybe we should delete this section and instead point to `setup.sh` where we would be careful to maintain explanatory comments.

Comment (imobachgs) [reply README.md:104]: I agree. In this case, it makes sense to put the documentation next to the code.

File: service/share/org.opensuse.DInstaller.service [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[D-BUS Service]
+Name=org.opensuse.DInstaller
+Exec=/usr/bin/d-installer manager
+User=root
+SystemdService=d-installer.service
```
Comment (mvidner) [inline service/share/org.opensuse.DInstaller.service:5]: Ha! Adding this line broke the startup:


> Nov 10 05:14:01 dhcp116.suse.cz dbus-daemon[701]: [system] Activating via systemd: service name='org.opensuse.DInstaller' unit='d-installer.service' requested by ':1.101' (uid=0 pid=3514 comm="cockpit-bridge")
Nov 10 05:14:01 dhcp116.suse.cz dbus-daemon[701]: [system] Activation via systemd failed for unit 'd-installer.service': Unit d-installer.service not found.

File: service/share/systemd.service [hunk 3]
```diff
@@ -6,7 +6,7 @@ After=network-online.target
 [Service]
 Type=dbus
 BusName=org.opensuse.DInstaller
-ExecStart=/usr/bin/d-installer
+ExecStart=/usr/bin/d-installer manager
```
Comment (imobachgs) [inline service/share/systemd.service:9]: :+1:

**Review by imobachgs — APPROVED** [review]

**Review by imobachgs — APPROVED** [review]
Great job! Now those init scripts do not feel like a temporary solution anymore :-)

File: service/share/org.opensuse.DInstaller.service (hunk 2 above)
Comment (mvidner) [reply service/share/org.opensuse.DInstaller.service:5]: (solved by installing also the systemd service in `build.sh`)

**Pull request merged**